service/controller/internal/changedetection/tcnp.go

```diff
@@ -93,6 +95,23 @@ func (t *TCNP) ShouldUpdate(ctx context.Context, md infrastructurev1alpha2.AWSMa
 		t.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("detected tenant cluster node pool should update due to operator version changes from %#q to %#q", cc.Status.TenantCluster.OperatorVersion, key.OperatorVersion(&md)))
 		return true, nil
 	}
+	if !securityGroupEqual {
+		t.logger.LogCtx(ctx, "level", "debug", "message", "detected tenant cluster node pool should update due to node pool security groups")
+		return true, nil
+	}
 
 	return false, nil
 }
+
+func securityGroupListEqual(s1 []*ec2.SecurityGroup, s2 []*ec2.SecurityGroup) bool {
+	if len(s1) != len(s2) {
+		return false
+	}
+	for i := 0; i < len(s1); i++ {
+		if s1[i].GroupId != s2[i].GroupId {
```

Comment:
I am not certain if EC2 api returns the groups always in the same order or if  this can be random order

Reply:
We should code defensive and consider the order is random.